### PR TITLE
Support missing BQ write parameters

### LINKED ISSUE #3928: Support clustering in saveAs..BigQueryTable
I am trying to write to a clustered table using `saveAsTypedBigQueryTable` and getting:

```
<table> is configured to use TableDestinationCoderV2. Set withClustering() on BigQueryIO.write() and,  if you provided a custom DynamicDestinations instance, override getDestinationCoder() to return TableDestinationCoderV3.
```

I only hit this when I am processing many partitions at once. I see that the SCIO API doesn't expose the withClustering option.

https://beam.apache.org/releases/javadoc/2.31.0/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.Write.html#withClustering-com.google.api.services.bigquery.model.Clustering-

### LINKED ISSUE #4334: Support BigQuery Load jobs for batch loading
From beam [documentation](https://beam.apache.org/documentation/io/built-in/google-bigquery/#setting-the-insertion-method) 

> When you specify load jobs as the insertion method using BigQueryIO.write().withMethod(FILE_LOADS).

Scio should also give users the possibility to use files to load data to BigQuery

### PR DESCRIPTION
**Pull request by RustedBones**
Fix #4334 
Fix #3928

Part of #3332

### CODE REVIEW

**Comment by amo12937** [comment]
@RustedBones 

Thank you for approving & merging my PR.

Is there a plan to merge #4810 into `main` branch? I'd like to use #4813 in my project.
Or can I contribute anything in this PR so that it will be merged? Or should I open the same PR as #4813 to `main`?

I apologize for my poor English.

Thank you.

**Comment by RustedBones** [comment]
Hi @amo12937,
Yes, this PR is planned for the `0.13.0` milestone (I forgot to add the label). We're planing to release it ASAP.
The PR is in draft because some BQ options are still missing. Will convert to ready when this is done, probably during the week.

**Comment by amo12937** [comment]
@RustedBones 
Thank you for replying me. I appreciate your work. I'm looking forward to `0.13.0`.

File: scio-google-cloud-platform/src/main/scala/com/spotify/scio/bigquery/BigQueryIO.scala [hunk 1]
```diff
@@ -215,41 +245,50 @@ object BigQueryTypedTable {
...
+      schema: TableSchema = DefaultSchema,
+      writeDisposition: WriteDisposition = DefaultWriteDisposition,
+      createDisposition: CreateDisposition = DefaultCreateDisposition,
+      tableDescription: String = DefaultTableDescription,
+      timePartitioning: TimePartitioning = DefaultTimePartitioning,
+      clustering: Clustering = DefaultClustering,
+      triggeringFrequency: Duration = DefaultTriggeringFrequency,
+      sharding: Sharding = DefaultSharding,
+      failedInsertRetryPolicy: InsertRetryPolicy = DefaultFailedInsertRetryPolicy,
+      configOverride: ConfigOverride[T] = defaultConfigOverride
```
Comment (RustedBones) [inline scio-google-cloud-platform/src/main/scala/com/spotify/scio/bigquery/BigQueryIO.scala:276]: Is the `configOverride` naming convention OK, and with default as `identity` ?

File: scio-test/src/main/scala/com/spotify/scio/testing/TransformFinder.scala [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+package com.spotify.scio.testing
+
+import org.apache.beam.sdk.Pipeline.PipelineVisitor
+import org.apache.beam.sdk.Pipeline.PipelineVisitor.CompositeBehavior
+import org.apache.beam.sdk.runners.{PTransformMatcher, TransformHierarchy}
+import org.apache.beam.sdk.transforms.PTransform
+
+import scala.collection.mutable
+
+class TransformFinder(matcher: PTransformMatcher) extends PipelineVisitor.Defaults {
```
Comment (kellen) [inline scio-test/src/main/scala/com/spotify/scio/testing/TransformFinder.scala:26]: Some scaladoc wouldn't hurt. Not quite sure what `freed` means in this context... non-matching?

**Pull request merged**